lib/RecommendationProvider.js

```diff
@@ -48,13 +48,13 @@ RecommendationProvider.prototype = {
     * Handle the NEW_RECOMMENDATION event
     */
   _handleNewRecommendationEvent(currentRecommendation) {
-    let url = currentRecommendation ? currentRecommendation.url : null;
-    let recommenderType = currentRecommendation ? currentRecommendation.recommender_type : null;
-    this._tabTracker.handleUserEvent({
-      event: "NEW_RECOMMENDATION",
-      url: url,
-      recommender_type: recommenderType
-    });
+    if (currentRecommendation && currentRecommendation.url &&
+      currentRecommendation.recommender_type) {
+      let payload = {event: "NEW_RECOMMENDATION"};
+      payload.url = currentRecommendation.url;
+      payload.recommender_type = currentRecommendation.recommender_type;
```

Comment:
nit: is there a reason `url` and `recommender_type` are not part of the `payload` object when you first generate it? I think it should be safe to do that since you added that if statement right?